command/agent/config.go

```diff
@@ -1089,6 +1093,57 @@ func (n *NormalizedAddrs) Copy() *NormalizedAddrs {
 	return &nn
 }
 
+// SocketConfig contains the path and configuration for a unix domain socket
+// listener.
+type SocketConfig struct {
+	Path  string `hcl:"path"`
+	User  string `hcl:"user"`
+	Mode  string `hcl:"mode"`
+	Group string `hcl:"group"`
+}
+
+func (s *SocketConfig) Copy() *SocketConfig {
+	if s == nil {
+		return nil
+	}
+	ns := *s
+	return &ns
+}
+
+// Merge merges two SocketConfigs together, preferring values from the argument.
+func (s *SocketConfig) Merge(b *SocketConfig) *SocketConfig {
+	if b == nil {
+		return s
+	}
+	var result SocketConfig
+	if s != nil {
+		result = *s
+	}
```

Comment:
We just declared `Copy()` above, let's use that instead of implementing it twice